src/Provider/FileProvider.php

```diff
@@ -378,7 +385,20 @@ protected function doTransform(MediaInterface $media): void
      */
     protected function setFileContents(MediaInterface $media, ?string $contents = null): void
     {
-        $file = $this->getFilesystem()->get(sprintf('%s/%s', $this->generatePath($media), $media->getProviderReference()), true);
+        $providerReference = $media->getProviderReference();
+
+        if (null === $providerReference) {
+            throw new \RuntimeException(sprintf(
```

Comment:
I also throw an exception here, it should be consistent with the other providerReference cases.

Reply:
I would consider this as an InvalidArgument Exception